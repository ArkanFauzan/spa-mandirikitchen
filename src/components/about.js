import React from "react"
import {FormAbout} from "./about/form"

export class AboutComponent extends React.Component{
    constructor(){
        super()
        this.state={
            nama:"",
            email:"",
            pesan:"",
        }
    }

    _handleChange = (value) => {
        this.setState({[value.name]:value.value})
    }

    render(){

        const {nama, email, pesan} = this.state

        return(
            <div className="container">
                <img style={{display:"block", margin:"auto"}} width="100%"  src="img/cv-crop.jpg" />
                <h2 className="my-4">CV Mandiri Kitchen Perkasa</h2>
                <p className="mb-4" style={{fontSize:"20px",textAlign:"justify"}}>
                    Mandiri Kitchen Perkasa adalah sebuah perusahaan yang bergerak dibidang 
                    disain dapur restoran serta pembuatan peralatan dapur komersial. Disain 
                    atau perencanaan dapur restoran sangat diperlukan agar ruang dapur 
                    restoran anda menjadi efisien dan fungsional serta biaya yang dikeluarkan 
                    sesuai dengan hasil yg diinginkan karena Penataan Peralatan Memasak yang 
                    TEPAT dan BENAR.  Ruang lingkup dan spesifikasi pekerjaan kami 
                    meliputi Desain dan perencanaan dapur komersial, Pembuatan dan pengadaan 
                    alat alat dapur komersial, instalasi ducting dan instalasi gas restoran. 
                    Yang dimaksud Dapur Komersial disini adalah dapur yang dipakai untuk 
                    keperluan usaha, seperti Dapur Restoran, Dapur Hotel, Dapur Asrama maupun 
                    Dapur Catering</p>

                <h2>Kontak kami</h2>
                <table className="my-4">
                    
                    <tr>
                        <td>Gunadi</td>
                        <td>&nbsp;&nbsp;&nbsp;: 0821 xxxx xxxx</td>
                    </tr>
                    <tr>
                        <td>Heri</td>
                        <td>&nbsp;&nbsp;&nbsp;: 0812 xxxx xxxx</td>
                    </tr>
                    <tr>
                        <td>Website</td>
                        <td>&nbsp;&nbsp;&nbsp;: mandirikitchenperkasa.com</td>
                    </tr>
                    <tr>
                        <td>Email</td>
                        <td>&nbsp;&nbsp;&nbsp;: mandirikitchenperkasa@gmail.com</td>
                    </tr>
                    <tr>
                        <td>Alamat</td>
                        <td>&nbsp;&nbsp;&nbsp;: Depok, Jawa Barat</td>
                    </tr>
                </table>

                <h2>Atau melalui form berikut:</h2>

                {/* menampilkan controlled form dengan prinsip lifting state up */}
                <FormAbout state={this.state} onChangeValue={this._handleChange}/>
            </div>
        )
    }
}