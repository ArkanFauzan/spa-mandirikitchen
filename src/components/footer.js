import React from "react"

export class FooterComponent extends React.Component{
    render(){
        return(
            <>
                <div>
                    <br/>
                    <h3>Hubungi kami</h3>
                    0812 xxxx xxxx (A.N. Gunadi) <br/>0821 xxxx xxxx (A.N. Heri)
                </div><br/>
                <div style={{borderTop:"solid 1px black"}}>
                    <br/>
                    Mandiri Kitchen 2020 all rights reserved
                </div><br/>
            </>
        )
    }
}