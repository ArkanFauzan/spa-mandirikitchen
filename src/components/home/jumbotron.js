import React from "react"
import { Alert, Card, Button
} from 'react-bootstrap';

export function JumbotronComponenet(){
    return(
        <>
            <img src="/img/cv-crop.jpg" width="100%" style={{position:"sticky",top:"0",zIndex:"-1"}} />
            <div style={{backgroundColor:"white"}}>
                <Alert variant="light text-dark">
                    <h1 style={{textAlign:"center"}}>CV Mandiri Kitchen Perkasa</h1>
                    <h2 style={{textAlign:"center"}}>PERALATAN DAPUR RESTORAN</h2>
                </Alert>
                <img style={{display:"block", margin:"auto"}} width="700px" src="img/our-service.jpg" />
            </div>
        </>
    )
}