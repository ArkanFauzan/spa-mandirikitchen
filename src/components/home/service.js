import React from "react"
import { Card} from 'react-bootstrap';

const service = [
    {
        img:"img/24-7.jpg",
        title:<h3>Full Support & Fast Response</h3>,
        text:<>0812 xxxx xxxx (A.N. Gunadi) <br/>0821 xxxx xxxx (A.N. Heri)</>

    },
    {
        img:"img/quality.png",
        title:<><h3>Kualitas Terjaga</h3><br/></>,
        text:"Kami berkomitmen untuk memberikan produk terbaik"
        
    },
    {
        img:"img/free.jpg",
        title:<><h3>Free Konsultasi, Survei, dan Design </h3></>,
        text:"Kepuasan pelanggan adalah yang utama"
        
    }
];

// membuat list service
export const listService = service.map((value,index)=>{
    return (
        <div key={index} style={{textAlign:"center"}} className="col-12 col-lg-4 my-2">
            <Card style={{ width: '18rem', margin:'auto' }}>
                <Card.Img variant="top" width="286px" height="180px" src={value.img} />
                <Card.Body>
                    <Card.Title>{value.title}</Card.Title>
                    <Card.Text>
                       {value.text}
                    </Card.Text>
                </Card.Body>
            </Card>
        </div>
    )
})