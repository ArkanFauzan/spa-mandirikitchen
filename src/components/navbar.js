import React from 'react';
import { Navbar, NavDropdown, Button,
    Nav, Form, FormControl
} from 'react-bootstrap';

import {BrowserRouter as Router,
    Switch,
    Route,
    Link
} from 'react-router-dom'

export function NavbarComponent(){
    return(
        <Navbar fixed="top" bg="light" expand="lg">
            <Navbar.Brand ><Link to="/home"><img style={{marginRight:"50px",}} width="150px" src="img/logo-crop.jpg"/></Link></Navbar.Brand>
            <Navbar.Toggle aria-controls="basic-navbar-nav" />
            <Navbar.Collapse id="basic-navbar-nav">
              <Nav className="mr-auto">
                <Nav.Link style={{marginRight:"50px",fontWeight:"bold",textDecoration:"none"}}><Link to="/home">HOME</Link></Nav.Link>
                <Nav.Link style={{marginRight:"50px",fontWeight:"bold",textDecoration:"none"}}><Link to="/products">PRODUCTS</Link></Nav.Link>
                <Nav.Link style={{marginRight:"50px",fontWeight:"bold",textDecoration:"none"}}><Link to="/about">ABOUT</Link></Nav.Link>
              </Nav>
              <Form inline>
                <FormControl type="text" placeholder="Search" className="mr-sm-2" />
                <Button variant="outline-success">Search</Button>
              </Form>
            </Navbar.Collapse>
          </Navbar>
    )
}