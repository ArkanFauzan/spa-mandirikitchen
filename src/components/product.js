import React,{useState, useEffect, useContext} from "react" 
import {BarangContext} from "../barangContext"

// import list barang
import {ListBarang} from "./product/listbarang"

export function ProductComponent(){

    // menggunakan hooks untuk memfilter kategori
    const [state, Filter] = useState({filter:"semua kategori"}) ;
    const {filter} = state;

    // mengambil nilai barang dari provider (menggunakan Context API)
    const {barang} = useContext(BarangContext);

    // mengubah state filter
    const _handleChange = (event) => {
        Filter({filter:event.target.value})
    }

    // similar componentDidUpdate(){}
    useEffect(()=>{
        alert(`Produk yang ditampilkan berdasarkan kategori ${filter}`)
    },[filter])

    return(
        <div className="container-fluid" style={{marginTop:"100px"}}>
            <form>
                <label><p style={{fontSize:"25px",textAlign:"justify"}}>Tampilkan berdasarkan : </p></label>
                <select  style={{fontSize:"25px",padding:"7px 15px"}} name="filter" value={filter} onChange={_handleChange}>
                    <option value="semua kategori">Semua Kategori</option>
                    <option value="area memasak">Area Memasak</option>
                    <option value="area mencuci">Area Mencuci</option>
                </select>
            </form>

            {/* menampilkan list barang sesuai filter kategori */}
            <ListBarang barang={barang} filter={filter}/>
        </div>
    )
}