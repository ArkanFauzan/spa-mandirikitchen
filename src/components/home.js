import React from "react"
import { Alert, Card, Button
} from 'react-bootstrap';

import {Link, BrowserRouter as Router} from 'react-router-dom'
import {listService} from "./home/service"
import {JumbotronComponenet} from "./home/jumbotron"

export class HomeComponent extends React.Component{
    render(){
        return(
            <Router>
            {/* jumbotron */}
            <JumbotronComponenet />
            
            {/* Card Our service */}
            <div className="row pt-2 bg-light" style={{textAlign:'center'}}>
                {/* menampilkan list service */}
                {listService}

                <div className="my-4 col-12">
                    <Button variant="primary" size="lg" style={{margin:'auto'}}><Link style={{color:'white',textDecoration:'none'}} to="products">Let's See Our Product -&gt;</Link></Button>
                </div>
            </div>

            </Router>
        )
    }
}