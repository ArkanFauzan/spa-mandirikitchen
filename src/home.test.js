import React from 'react';
import {HomeComponent} from './components/home';
import renderer from 'react-test-renderer';

it('Home is renders correctly', () => {
  const tree = renderer
    .create(<HomeComponent/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});