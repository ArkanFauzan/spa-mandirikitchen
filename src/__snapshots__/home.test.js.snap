// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home is renders correctly 1`] = `
Array [
  <img
    src="/img/cv-crop.jpg"
    style={
      Object {
        "position": "sticky",
        "top": "0",
        "zIndex": "-1",
      }
    }
    width="100%"
  />,
  <div
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <div
      className="fade alert alert-light text-dark show"
      role="alert"
    >
      <h1
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        CV Mandiri Kitchen Perkasa
      </h1>
      <h2
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        PERALATAN DAPUR RESTORAN
      </h2>
    </div>
    <img
      src="img/our-service.jpg"
      style={
        Object {
          "display": "block",
          "margin": "auto",
        }
      }
      width="700px"
    />
  </div>,
  <div
    className="row pt-2 bg-light"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <div
      className="col-12 col-lg-4 my-2"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="card"
        style={
          Object {
            "margin": "auto",
            "width": "18rem",
          }
        }
      >
        <img
          className="card-img-top"
          height="180px"
          src="img/24-7.jpg"
          width="286px"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            <h3>
              Full Support & Fast Response
            </h3>
          </div>
          <p
            className="card-text"
          >
            0812 xxxx xxxx (A.N. Gunadi) 
            <br />
            0821 xxxx xxxx (A.N. Heri)
          </p>
        </div>
      </div>
    </div>
    <div
      className="col-12 col-lg-4 my-2"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="card"
        style={
          Object {
            "margin": "auto",
            "width": "18rem",
          }
        }
      >
        <img
          className="card-img-top"
          height="180px"
          src="img/quality.png"
          width="286px"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            <h3>
              Kualitas Terjaga
            </h3>
            <br />
          </div>
          <p
            className="card-text"
          >
            Kami berkomitmen untuk memberikan produk terbaik
          </p>
        </div>
      </div>
    </div>
    <div
      className="col-12 col-lg-4 my-2"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="card"
        style={
          Object {
            "margin": "auto",
            "width": "18rem",
          }
        }
      >
        <img
          className="card-img-top"
          height="180px"
          src="img/free.jpg"
          width="286px"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            <h3>
              Free Konsultasi, Survei, dan Design 
            </h3>
          </div>
          <p
            className="card-text"
          >
            Kepuasan pelanggan adalah yang utama
          </p>
        </div>
      </div>
    </div>
    <div
      className="my-4 col-12"
    >
      <button
        className="btn btn-primary btn-lg"
        disabled={false}
        style={
          Object {
            "margin": "auto",
          }
        }
        type="button"
      >
        <a
          href="/products"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
          Let's See Our Product -&gt;
        </a>
      </button>
    </div>
  </div>,
]
`;
